### PR TITLE
[CALCITE-6212] Config locale = 'en_US' for javadoc task

### PR DESCRIPTION
**Pull request by libenchao**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -486,6 +486,8 @@ allprojects {
             (options as StandardJavadocDocletOptions).apply {
                 // Please refrain from using non-ASCII chars below since the options are passed as
                 // javadoc.options file which is parsed with "default encoding"
+                // locale should be placed at the head of any options: https://docs.gradle.org/current/javadoc/org/gradle/external/javadoc/CoreJavadocOptions.html#getLocale
+                locale = "en_US"
```
Comment (snuyanzin) [inline build.gradle.kts:490]: Nice finding
Am I correct that same thing should be done for Avatica?

**Review by asolimando — APPROVED** [review]
LGTM

**Comment by JiajunBernoulli** [comment]
Is it better to register the affected version with Jira?

File: build.gradle.kts (hunk 1 above)
Comment (libenchao) [reply build.gradle.kts:490]: Yes, I do think so. I'll do this for avatica once this is merged.

**Comment by libenchao** [comment]
> Is it better to register the affected version with Jira?

I've logged CALCITE-6212, thanks~

File: build.gradle.kts (hunk 1 above)
Comment (libenchao) [reply build.gradle.kts:490]: The PR against avatica is now available for review: https://github.com/apache/calcite-avatica/pull/235

**Pull request closed without merging**